scopes/workspace/variants/variants.docs.md

```diff
@@ -0,0 +1,127 @@
+---
+id: variants
+title: Variants
+slug: /aspects/variants
+---
+
+The `teambit.workspace/variants` extension enables a cascading, CSS-like, selection of components in the workspace configuration. It provides order and simplicity to the way we apply rules and settings to large number of independent components - all in a single workspace.
```

Comment:
one of the most important things to mention here is how this merge between different rules is working:
we took aspects from each level and add them, but once we see an aspect we won't merge it's configuration if it's defined in an upper level again with different config (let's talk about the rational of it, because many developers already asks me why it's like this)
so for example:

`workspace.jsonc`
```json
{
  "teambit.workspace/variants": {
    "*": {
        "my-aspect1": {
          "aspect1-root-key": "aspect1-root-val"
        },
       "my-aspect2": {
        "aspect2-root-key": "aspect2-root-val"
      }
        "my-aspect4": {
          "aspect4-root-key": "aspect4-root-val"
        }
    },
    "components": {
      "my-aspect1": {
        "aspect1-components-key": "aspect1-components-val"
      },
      "my-aspect2": {
        "aspect2-components-key": "aspect2-components-val"
      }
    },
    "components/ui": {
      "my-aspect1": {
        "aspect1-components-ui-key": "aspect1-components-ui-val"
      },
      "my-aspect3": {
        "aspect3-components-ui-key": "aspect3-components-ui-val"
      }
    }
  }
}
```
`components/ui/button final config`
```json
{
  "my-aspect1": {
    "aspect1-components-ui-key": "aspect1-components-ui-val"
  },
  "my-aspect2": {
    "aspect2-components-key": "aspect2-components-val"
  },
  "my-aspect3": {
    "aspect3-components-ui-key": "aspect3-components-ui-val"
  },
  "my-aspect4": {
    "aspect4-root-key": "aspect4-root-val"
  }
}
```

As you can see, we didn't took the `aspect1-components-ui-key` nor the `aspect1-root-key` (because we already saw `my-aspect1` in a more specific rule.
We also didn't took the `aspect2-root-key` because we already saw it in the `components` rule